sympy/geometry/util.py

```diff
@@ -310,6 +312,173 @@ def _orientation(p, q, r):
     return Polygon(*convexHull)
 
 
+def closest_points(*args):
+    """The closest pair of points among a given set of points in 2D plane.
+
+    Parameters
+    ==========
+
+    args : a collection of Points on 2D plane.
+
+    Returns
+    =======
+
+    pair : A pair of closest points
+
+    Notes
+    =====
+
+    This can only be performed on a set of non-symbolic points.
+
+    References
+    ==========
+
+    [1] http://www.cs.mcgill.ca/~cs251/ClosestPair/ClosestPairPS.html
+
+    [2] Sweep line algorithm
+    https://en.wikipedia.org/wiki/Sweep_line_algorithm
+
+    Examples
+    ========
+
+    >>> from sympy.geometry import closest_points, Point2D
+    >>> points = [Point2D(5, 2), Point2D(7, 6), Point2D(3, 9), Point2D(9, 4), Point2D(8, 5)]
+    >>> closest_points(*points)
+    (Point2D(8, 5), Point2D(7, 6))
+
+    >>> from sympy.geometry import closest_points, Point2D
+    >>> points = [Point2D(80, 89), Point2D(11, 85), Point2D(32, 51), Point2D(59, 59), Point2D(39, 80), Point2D(62, 73)]
+    >>> closest_points(*points)
+    (Point2D(62, 73), Point2D(59, 59))
+
+    """
+    from collections import deque
+    from math import sqrt, pow
```

Comment:
I'm not sure what is going on but apparently the change of size or scale effected by sqrt matters. It would be interesting to know if other functions such as log would have a similar effect.